### PR TITLE
add webextension-polyfill

### PR DESCRIPTION
**Pull request by AprilSylph**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "x8",
+  "version": "0.0.1",
+  "homepage": "https://new-xkit-extension.tumblr.com/",
+  "license": "GPL-3.0",
```
Comment (nightpool) [inline package.json]: the license in the repo is AGPL-3.0-or-later iirc

File: package.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "browser": "inject.js",
+  "devDependencies": {
+    "webextension-polyfill": "^0.6.0",
+    "web-ext": "^4.0.0"
```
Comment (nightpool) [inline package.json:13]: what's web-ext used for?

**Review by nightpool — CHANGES_REQUESTED** [review]
what's the process to get the browser-polyfill file into vendor/? mind adding a small script for it?

**Comment by nightpool** [comment]
also: yarn lockfile/package-lock.json?

File: package.json (hunk 2 above)
Comment (AprilSylph) [reply package.json:13]: building for firefox (as specified in the readme :P)

**Comment by AprilSylph** [comment]
i'm not sure how best to add a script, the process is just `cp node_modules/webextension-polyfill/dist/browser-polyfill.min.js vendor/`. i would've used the file directly but i assume we don't want to package `node_modules` with the extension

**Comment by AprilSylph** [comment]
~~also i do not know how to create package-lock.json~~
nevermind i figured it out. i was running npm 3

**Comment by nightpool** [comment]
@AprilSylph can you add at least a simple npm script to copy the file?

**Comment by AprilSylph** [comment]
@nightpool we'll need an extra dependency for it to work across operating systems but sure

**Review by nightpool — APPROVED** [review]

**Pull request merged**